ipatests/test_integration/test_caless.py

```diff
@@ -1507,9 +1507,91 @@ def install(cls, mh):
 
     @replica_install_teardown
     def test_server_replica_install_pkinit(self):
+        "Install master and replica with PKINIT"
+
         self.create_pkcs12('ca1/replica', filename='replica.p12')
         self.create_pkcs12('ca1/replica-kdc', filename='replica-kdc.p12')
         result = self.prepare_replica(pkinit_pkcs12_exists=True,
                                       pkinit_pin=_DEFAULT)
         assert result.returncode == 0
         self.verify_installation()
+
+
+class TestServerCALessToCAFull(CALessBase):
+    num_replicas = 0
+
+    @classmethod
+    def install(cls, mh):
+        super(TestServerCALessToCAFull, cls).install(mh)
+
+        cls.create_pkcs12('ca1/server')
+        cls.prepare_cacert('ca1')
+        master = cls.install_server()
+        assert master.returncode == 0
+
+    @server_install_teardown
+    def test_server_ipa_ca_install(self):
+        "Install CA on master"
+
+        ca = tasks.install_ca(self.master)
+        assert ca.returncode == 0
+
+        cert_update = self.master.run_command(['ipa-certupdate'])
+        assert cert_update.returncode == 0
+
+
+class TestReplicaCALessToCAFull(CALessBase):
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        super(TestReplicaCALessToCAFull, cls).install(mh)
+
+        cls.create_pkcs12('ca1/server')
+        cls.prepare_cacert('ca1')
+        master = cls.install_server()
+        assert master.returncode == 0
+
+    @replica_install_teardown
+    def test_replica_ipa_ca_install(self):
+        "Install CA on replica (master caless)"
+
+        self.create_pkcs12('ca1/replica', filename='replica.p12')
+
+        result = self.prepare_replica()
+        assert result.returncode == 0
+        ca_replica = tasks.install_ca(self.replicas[0])
+        assert ca_replica.returncode == 0
+        cert_update_replica = self.replicas[0].run_command(['ipa-certupdate'])
+        assert cert_update_replica.returncode == 0
+
+
+class TestServerReplicaCALessToCAFull(CALessBase):
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        super(TestServerReplicaCALessToCAFull, cls).install(mh)
+
+        cls.create_pkcs12('ca1/server')
+        cls.prepare_cacert('ca1')
+
+        master = cls.install_server()
+        assert master.returncode == 0
+
+    @replica_install_teardown
+    def test_server_replica_ipa_ca_install(self):
```

Comment:
You are right, moreover it looks like TestServerReplicaCALessToCAFull is already doing so.